### PR TITLE
Add some constraints around signature algorithms

### PR DESCRIPTION
**Pull request by siyengar**
Add some constraints around signature algorithms

### CODE REVIEW

File: draft-rescorla-tls-subcerts.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ The signature on the sub-certificate indicates a delegation from the
 master certificate which is issued to the TLS server operator. The key pair used
 to sign a sub-certificate is presumed to be one whose public key is
 contained in an X.509 certificate that associates one or more names to
-the sub-certificate signing key.
+the sub-certificate signing key.
```
Comment (ekr) [inline draft-rescorla-tls-subcerts.md]: You may want to update your editor to avoid trailing ws :)

File: draft-rescorla-tls-subcerts.md [hunk 2]
```diff
@@ -97,6 +97,11 @@ few important ways:
 * The client uses the public key in the sub-certificate as the server's
   working key for the TLS handshake.
 
+The sub-certificate signature is subject to the negotiated signature algorithms.
+A sub-certificate cannot be used if the client advertises support for sub-certificate
+however a server does not have a leaf certificate which is compatible with any of  
+the negotiated signature algorithms.
+
```
Comment (ekr) [inline draft-rescorla-tls-subcerts.md:104]: By "leaf" you mean what I have been calling master?

File: draft-rescorla-tls-subcerts.md (hunk 1 above)
Comment (siyengar) [inline draft-rescorla-tls-subcerts.md]: twas github :), will remove

File: draft-rescorla-tls-subcerts.md (hunk 2 above)
Comment (siyengar) [inline draft-rescorla-tls-subcerts.md:104]: ya will update the language to say master instead

**Pull request merged**